util/top_recommendations.py

```diff
@@ -0,0 +1,39 @@
+import bisect
+
+
+class TopRecommendations(object):
+
+    def __init__(self, n_recommendations):
+        self.recommendations_values = []
+        self.recommendations_indices = []
+        self.n_recommendations = n_recommendations
+
+    def insert(self, index, value):
+        if self.n_recommendations > len(self.recommendations_values):
+            inserted_index = self._insert_and_return_index(self.recommendations_values, value)
+            self.recommendations_indices = self._insert_at_index(self.recommendations_indices, index, inserted_index)
+            return
+        if len(self.recommendations_values) != 0:
+            if self.recommendations_values[0] < value:
+                self.recommendations_values.pop(0)
+                self.recommendations_indices.pop(0)
+                inserted_index = self._insert_and_return_index(self.recommendations_values, value)
+                self.recommendations_indices = self._insert_at_index(self.recommendations_indices,
+                                                                     index, inserted_index)
+
+    def _insert_and_return_index(self, arr, val):
+        index = bisect.bisect(arr, val)
+        bisect.insort(arr, val)
+        return index
+
+    def _insert_at_index(self, arr, val, index):
+        return arr[0: index] + [val] + arr[index:]
```

Comment:
bisect is a python module, and yes they have insert at index but the insert is not sorted. And if it doesn't matter if the complexity is quadratic, the prediction array is sparse which means most of the inserts actually won't make it to this function